components/channel_invite_modal/channel_invite_modal.jsx

```diff
@@ -157,6 +159,8 @@ export default class ChannelInviteModal extends React.Component {
             rowSelected = 'more-modal__row--selected';
         }
 
+        this.props.actions.getStatus(option.id);
```

Comment:
We shouldn't be triggering side effects like loading statuses in part of the `render` method. React makes some assumptions that these methods have no side effects to optimize some of what it does under the hood, but more importantly, this will cause `getStatus` to be called any time this component re-renders. If I'm reading this correctly, this will actually cause it to be called repeatedly as long as this is visible because the result of `getStatus` will cause `this.props.userStatuses` to change which will make this re-render and trigger over and over again.

The proper way to do this would be to look at the results of `getProfilesNotInChannel` and `searchProfiles` when they're called to see which users are missing statuses and then using [`getStatusesByIds`](https://github.com/mattermost/mattermost-redux/blob/master/src/actions/users.js#L698) to load those all at once.

Also, if you were to add an action to get the missing statuses for a given list of users, that'd be awesome, but it's not strictly necessary since that would involve also involve changes to mattermost-redux.